pcdsdevices/lens.py

```diff
@@ -286,31 +341,294 @@ def _make_safe(self):
             filt.insert()
             time.sleep(0.01)
         if filt.inserted:
-            print("REMINDER: Beam stop attenuator moved in!")
+            logger.info('Beam stop attenuator moved in!')
             safe = True
         else:
-            print("WARNING: Beam stop attenuator did not move in!")
+            logger.warning('Beam stop attenuator did not move in!')
             safe = False
         return safe
 
 
 class LensStack(LensStackBase):
+    """
+    Class for Be lens.
+
+    Parameters
+    ----------
+    x_prefix : str
+        The EPICS prefix that identifies the x motor.
+    y_prefix : str
+        The EPICS prefix that identifies the y motor.
+    z_prefix : str
+        The EPICS prefix that identifies the z motor.
+    lens_set : list, optional
+        List of lens sets. e.g. [numer1, lensthick1, number2, lensthick2...]
+    z_offset : number
+        Distance from sample to lens_z=0 in meters.
+    z_dir : number
+        1 or -1, represents beam direction wrt z direction.
+    E: number, optional
+        Beam energy
+    att_obj : attenuator object, optional
+    lcls_obj
+        Object that gets PVs from lcls (for energy)
+    mono_obj
+        Object that gets energy from monochromator
+    beamsize_unfocused : float, optional
+        Radial size of x-ray beam before focusing.
+    path : str
+        Path to the file that defines which lenses are being used.
+
+    Examples
+    --------
+
+    Before using the LclsStack class, provide the path of the be lens set file
+    to be used for the calculations.
+
+    If no lens sets are added in the file yet, use the
+    `be_lens_calcs.set_lens_set_to_file` function to set the lens sets:
+
+    >>> import pcdsdevices.lens as lens
+    >>> import pcdscalc.be_lens_calcs as be
+
+    >>> path = '../path/to/lens_set'
+
+    >>> sets_list = [[3, 0.0001, 1, 0.0002],
+                     [1, 0.0001, 1, 0.0003, 1, 0.0005],
+                     [2, 0.0001, 1, 0.0005]]
+    >>> be.set_lens_set_to_file(sets_list, path)
+
+    Create the LensStack() object by providing the x, y and z `prefixes`, as
+    well as all the other parameters:
+
+    >>> att = Attenuator('att', 4, name='att')
+
+    >>> be_stack = lens.LensStack(path=path, x_prefix='X:PREF',
```

Comment:
when we create the LensStack object, do we get the Energy  coming from? In the old code we have something like this:
https://github.com/pcdshub/old-hutch-configs/blob/18c7769d66a647ab558ad7ebc34d019fc98fbf48/xpp/blinst/crl.py#L102

Is this a good place to apply what we discussed here:
https://github.com/pcdshub/pcdscalc/pull/8#discussion_r509793381

Basically create the subscription in the LCLS object  to get the right E and have LensStack use that info?

Reply:
I'm not immediately sure how to implement here- maybe we need to register the `BeamStats` and/or `LCLS` objects that are created and have methods that set up the subscription callbacks? Maybe a little more thought needed.

Reply:
I am not sure how to deal with this, it seems like there are two pieces to it:

In Crl old object we have something like this:
https://github.com/pcdshub/old-hutch-configs/blob/18c7769d66a647ab558ad7ebc34d019fc98fbf48/xpp/blinst/crl.py#L102-L136

where both the `monoObj` and `lclsObj` are involved in getting the Energy. The lclsObj would just be similar to what we said above, but with `monoObj` is a bit more involved...

Is the old `monoObj` - the current `LODCM` object?

Reply:
oh that's interesting

Yes the mono obj is the LODCM usually- except when it is the CCM... XPP and XCS each have two monochromators. So deciding which mono object to use is experiment-specific... and the lodcm calcs aren't even set up yet in the python 3 libraries

Reply:
Ok we can use the LclsObject's energy for now, and then later we can add the LODCM's into play too - at least we should probably do this in a separate Issue - or what do you think?

Reply:
Yes I'd consider adding LODCM/CCM into this as a separate issue. In practice it should be OK to just override the energy with your planned energy for the experiment.